frontend/release-notes.jsx

```diff
@@ -0,0 +1,189 @@
+      .then(() => removeNote())
+      .catch((err) => alert(err.message));
+  };
+
+  if (note.releases.length === 1) {
+    return <p>Yes</p>;
+  }
+  return <input type="button" value="Make release-specific" onClick={makeReleaseSpecific} />;
+}
+
+function NoteRow({ note, removeNote, releaseApiUrl, converter }) {
+  return (
+    <tr>
+      <td style={cellStyles}>
+        <a href={`/admin/rna/note/${note.id}/`}>Edit</a>
+      </td>
+      <td style={cellStyles}>
+        {note.is_known_issue && note.is_known_issue !== releaseApiUrl ? "Known issue" : note.tag}
+      </td>
+      {/* biome-ignore lint/security/noDangerouslySetInnerHtml: Markdown is safe/trusted input */}
+      <td style={cellStyles} dangerouslySetInnerHTML={{ __html: converter.makeHtml(note.note) }} />
+      <td style={cellStyles}>
+        <BugLink bug={note.bug} />
+      </td>
+      <td style={cellStyles}>{note.sort_num}</td>
+      <td style={cellStyles}>
+        <ReleaseSpecific note={note} removeNote={removeNote} releaseApiUrl={releaseApiUrl} />
+      </td>
+      <td style={cellStyles}>
+        <input type="button" value="Remove" onClick={() => removeNote(note)} />
+      </td>
+    </tr>
+  );
+}
+
+function NoteRows({ data, removeNote, releaseApiUrl, converter }) {
+  return (
+    <tbody>
+      {data.map((note, index) => (
+        <NoteRow
+          key={note.id}
+          note={note}
+          removeNote={removeNote}
+          releaseApiUrl={releaseApiUrl}
+          converter={converter}
+        />
+      ))}
+    </tbody>
+  );
+}
+
+function NoteHeader({ data }) {
+  return (
+    <thead>
+      <tr>
+        {data.map((header) => (
+          <th key={header} style={cellStyles}>
+            {header}
+          </th>
+        ))}
+      </tr>
+    </thead>
+  );
+}
+
+function NoteTable({ url, releaseApiUrl, converter }) {
+  const [data, setData] = useState([]);
+
+  const getNotes = useCallback(() => {
+    fetch(url)
+      .then((res) => {
+        if (!res.ok) throw new Error(`HTTP error ${res.status}`);
+        return res.json();
+      })
+      .then(setData)
+      .catch((err) => alert(`Unable to get notes: ${err.message}`));
+  }, [url]);
+
+  const addNote = useCallback(
+    (id) => {
+      fetch(`/rna/notes/${id}/`)
+        .then((res) => {
+          if (!res.ok) throw new Error(`HTTP error ${res.status}`);
+          return res.json();
+        })
+        .then((note) => {
+          const releases = JSON.stringify({ releases: [...note.releases, releaseApiUrl] });
+          authPost(note.url, releases, true)
+            .then(getNotes)
+            .catch((err) => alert(err.message));
+        })
+        .catch((err) => alert(`Unable to add note: ${err.message}`));
+    },
+    [releaseApiUrl, getNotes],
+  );
+
+  const removeNote = (note) => {
+    const releases = JSON.stringify({
+      releases: note.releases.filter((url) => url !== releaseApiUrl),
+    });
+    authPost(note.url, releases, true)
```

Comment:
So this posts/patches the note resource with a new list of releases that it's a part of?

I ask because it looks (here and originally) like creating a note and deleting a note are just about which releases a note is associated with - is that right?

Reply:
Yeah, with DRF the related releases are the URLs to that release. So these calls update that list of URLs and DRF handles the updates.

Reply:
I added `authPatch` back b/c it made it a bit clearer which was POST and which was PATCH.